syntax = "proto3";

message CartItem {
  string product_id = 1;
  string product_name = 2;
  int32 quantity = 3;
  double price = 4;
}

message Fee {
    double total = 1;
    double platform_fee = 2;
    double surge = 3;
    double delivery = 4;
    double tax = 5;
    double discount = 6;
    double final = 7;
}

message Cart {
  string user_id = 1;
  repeated CartItem items = 2;
  Fee fee = 3;
  string coupon = 4;
  string status = 5;
  string created_at = 6; 
  string updated_at = 7;
}
